### ISSUE TITLE
kubectl top node should not have a column for storage

### ISSUE THREAD

**Issue by derekwaynecarr** [kubernetes/kubernetes#30782]
`kubectl top node` defines columns for cpu, memory, and storage.

I get cpu, and memory.

I don't get storage for either a pod or a node.

I think the column should be removed.

It's not clear if storage is rootfs, nodefs, or some other fs.

Absent a defined node spec for storage, and ability to provide quality of service around storage, we should not have the column.

@kubernetes/sig-node @vishh @davidopp @smarterclayton

**Comment by vishh** [comment]
Let's think from a scheduling standpoint. `disk` as a resource will most-likely apply to localhost `volumes`, containers and possibly pods. 
If `kubectl top` is expected to help determine available capacity or current utilization of a cluster, then we only need to consider the root partition. The image fs is a hidden implementation detail. If kubelet starts tracking volumes storage separately, then we might have to add sub-columns or find other means. Given that kubelet only recognized a single root partition, we can still expose its usage via `kubectl top`.
On the other hand, if `kubectl top` is expected to a cluster admins favorite tool, then we need to provide a lot more data on storage. I suspect that this falls under the monitoring charter.

In summary, if we can expose usage of root partition on all nodes, that might be a good start for v1.4

**Comment by derekwaynecarr** [comment]
Absent a clearly defined spec for 1.4, I thought the column should be
removed.  Otherwise it appears to be a bug.  I don't really understand how
imagefs is a hidden implementation detail.  It's not hidden to our
operators.

**Comment by vishh** [comment]
I'm not sure if `kubectl top` is meant for operators. Imagefs could be an
implementation detail. For example eviction thresholds are not exposed
currently via kubectl, which means that the actual utilization cannot
really grow up to available capacity.

**Comment by vishh** [comment]
In any case, if no values are going to show up for Storage in v1.4, I'm ok
with removing it.

**Comment by derekwaynecarr** [comment]
Fixed in https://github.com/kubernetes/kubernetes/pull/30767
